frontend/views/layout_head.html.erb

```diff
@@ -1,3 +1,31 @@
+<%
+
+def map_user_defined_to_labels(properties)
+  properties.map {|property| I18n.t("user_defined.#{property}")}
+end
+
+%>
+
 <% if AppConfig.has_key?(:user_defined_in_basic) %>
-  <%= javascript_include_tag "#{AppConfig[:frontend_prefix]}assets/user_defined_in_basic.js" %>
+    <%= javascript_include_tag "#{AppConfig[:frontend_prefix]}assets/user_defined_in_basic.js" %>
+
+    <script>
+     var record_type = '<%= controller.controller_name %>';
+     var fields = <%= ASUtils.to_json(map_user_defined_to_labels(AppConfig[:user_defined_in_basic].fetch(controller.controller_name, []))).html_safe %>;
+     var read_only_view = <%= controller.action_name == 'show' %>;
+     
+     var init = function () {
+         setTimeout(function () {
```

Comment:
had to use setTimeout here to push the initialization to the end of the process (to make sure the button handlers were in).  That was a low point too.  I should have added a comment for that.